### ISSUE TITLE
[TextField] Interactive elements non-text contrast

### ISSUE THREAD

**Issue by sarahill** [Shopify/polaris-react-archive#7977]
## Issue summary

Only the top border on the `TextField` meets the 3:1 contrast requirements for non-text elements. Since border is the only visual indicator of the control, the whole border needs to meet the 3:1 contrast requirement and not just the top border.

![text field contrast](https://user-images.githubusercontent.com/8629173/210418884-3b932343-87db-4768-b2a6-c45deffd59ea.png)

## Expected behavior

All four borders (not just the top border) should meet a minimum 3:1 contrast ratio with the page background or there should be another visual indicator to distinguish the control.

WCAG:
> Active User Interface Components
> For active controls any visual information provided that is necessary for a user to identify that a control is present and how to operate it must have a minimum 3:1 contrast ratio with the adjacent colors.
> ...
> Boundaries
> This success criterion does not require that controls have a visual boundary indicating the hit area, but if the visual indicator of the control is the only way to identify the control, then that indicator must have sufficient contrast.

https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html

It is a little vague, but my understanding is since the border is the only way we are distinguishing this control we need to add the boarder to all sides or add an additional visual indicator. This is not true of our elements such as `Button` or `Select`, as there are other elements to help distinguish them.

## Actual behavior

Only the top border on the `TextField` meets the 3:1 contrast requirements for non-text elements.

Related [PR](https://github.com/Shopify/polaris/pull/7893)
Related [GitHub issue](https://github.com/Shopify/polaris/issues/7838)

## Reported by
@RubenSandwich, feel free to ask him questions related to this ticket
